.github/workflows/run-e2e-tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Run E2E Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  go-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: copy E2E config
+        run: make copy_test_config
+
+      - name: update E2E config from secrets
+        env:
+          GATEWAY_PRIVATE_KEY: ${{ secrets.GatewayPrivateKey }}
```

Comment:
The secret name in the repo is all caps eg `GATEWAYPRIVATEKEY` so we either need to change it here or change the secret name to PascalCase.

Reply:
GH secrets are case-insensitive, but this is a good catch: updated to be consistent with `poktroll`.